build failure on Polaris

Out of the box the mochi-test repo fails to build on Polaris with the following message with the most recent software updates:

```
In file included from /soft/compilers/cudatoolkit/cuda-11.8.0/bin/../targets/x86_64-linux/include/cuda_runtime.h:83,
                 from <command-line>:
/soft/compilers/cudatoolkit/cuda-11.8.0/bin/../targets/x86_64-linux/include/crt/host_config.h:132:2: error: #error -- unsupported GNU version! gcc versions later than 11 are not supported! The nvcc flag '-allow-unsupported-compiler' can be used to override this version check; however, using an unsupported host compiler may cause compilation failure or incorrect run time execution. Use at your own risk.
```

These tests were developed when Polaris was still using gcc 11.x.  This is still our best test platform at the moment for CUDA support so it would be good to get this working again (maybe just using the unsupported compiler flag).

--- a/perf-regression/Makefile.subdir
+++ b/perf-regression/Makefile.subdir
@@ -9,7 +9,7 @@ if HAVE_GPU_TESTING
 bin_PROGRAMS += perf-regression/gpu-margo-p2p-bw
 
 #Remove -pthread doesn't work for nvcc
-CUDA_CFLAGS := $(subst -pthread,,$(CFLAGS))
+CUDA_CFLAGS := $(subst -pthread,,$(CFLAGS)) -allow-unsupported-compiler
 
 #Set up mpi path for include/lib
 MPICC := $(shell which $(CC))

